tools/integration_tests/implicitdir/implicitdir_test.go

```diff
@@ -130,6 +152,17 @@ func TestMain(m *testing.M) {
 	}
 
 	log.Printf("Test log: %s\n", logFile)
+
+	// tmp dir setup
+	tmpDir, err := os.MkdirTemp(mntDir, "tmpDir")
+	if err != nil {
+		printAndExit(fmt.Sprintf("Mkdir at %q: %v", mntDir, err))
+	}
+
+	// tmp file setup
+	fileName = path.Join(tmpDir, "tmpFile")
```

Comment:
if you are creating one tempFile and using it in all writeTests, then contents of the file will be modified after every test method execution. So in every test method, you might have to read the contents of tmpFile and then do string manipulation to do comparing with after the append. Just assuming "line1line2" in  every test will not work.

Another option would be to create tmpFile in every test method.

Reply:
Yes. I am aware of the contents of file being changed after every test hence have written the tests accordingly.

Reply:
your current test doesn't take care of it. Once you add more tests, we can't guarantee the execution order.

Reply:
Ohh. When I tested with multiple tests in the file, they were executed in the order they were declared. Is the general behaviour different from this?

Reply:
Then i.g the old code where the file was generated and written in every tests was correct?